Automattic/woocommerce-payments
# Add ExPlat PHP client

Fixes (partially) #1815

#### Changes proposed in this Pull Request

Adds a PHP client to interface with ExPlat A/B testing. This will be used to add the menu items in #1815.

This class is experimental because it is likely to be replaced. The end goal is to have this functionality available from WooCommerce Core, however it will be iterated here initially due to the shorter release cycle.

I have forked the `jetpack-attest` package and updated it with support for ExPlat. I am working with Jetpack to contribute these changes back. There is an issue for it in the Jetpack repo https://github.com/Automattic/jetpack/issues/19596. If those changes are accepted, we could switch over to using the package with Composer.

Automated tests were not added. The Jetpack package includes PHP tests but they are not compatible with the WC Pay’s tests due to different dependencies. The contribution back to Jetpack will have updated tests.

#### Testing instructions

Set up a site with Jetpack connected and tracking allowed.

Add the following snippet to `includes/class-wc-payments.php` on line 186: 

```
$abtest = new \WCPay\Experimental_Abtest(
	$_COOKIE[ 'tk_ai' ],
	'woocommerce',
	'yes' === get_option( 'woocommerce_allow_tracking' )
);

var_dump( $abtest->get_variation( 'wcpay_inapp_payments_connect_page_v2' ) );

die();
```

Load a page and see that the snippet above returns a string with your variation.

Reload the page to ensure caching works.

-------------------
